tests/baselines/reference/jsDeclarationsReusesExistingTypeAnnotations.js

```diff
@@ -0,0 +1,93 @@
+//// [index.js]
+class С1 {
+    /** @type {string=} */
+    p1 = undefined;
+
+    /** @type {string | undefined} */
+    p2 = undefined;
+
+    /** @type {?string} */
+    p3 = null;
+
+    /** @type {string | null} */
+    p4 = null;
+}
+
+class С2 {
+    /** @type {string=} */
+    get p1() {
+        return undefined;
+    }
+
+    /** @type {string | undefined} */
+    get p2() {
+        return undefined;
+    }
+
+    /** @type {?string} */
+    get p3() {
+        return null;
+    }
+
+    /** @type {string | null} */
+    get p4() {
+        return null;
+    }
```

Comment:
can you add tests for set and get/set pairs, the latter with annotations on each of get/set/get+set?

Reply:
@sandersn I've added tests